executor/executable/controllabletask.go

```diff
@@ -387,17 +392,35 @@ func (t *ControllableTask) Kill() error {
 			"targetList": cmd.TargetList,
 		}).
 		Debug("state DONE not reached, about to commit transition")
+		
+		// Call cmd.Commit() asynchronous
+		commitDone := make(chan *CommitResponse)
+    	go func() {
+			var cr CommitResponse
+			cr.newState, cr.transitionError = cmd.Commit()
+			commitDone <- &cr
+		}()
+		
+		// Set timeout cause OCC is locking up so killing is not possible. The following approach
+		// help us to bypass the OCC transition, so we can kill the tasks on force destroy env. 
+		// Currently this will run for every Kill message receive (force or not).
+		// TODO: Find a better way to distinguish force from plain Kill message.
```

Comment:
I actually think that this behavior is correct even if `force == false`. On the core side we distinguish between force or not, so we allow or disallow the termination depending on the global state when `force == false`.

But once we get to this `Kill()`, we already know that the preconditions for terminations were satisfied at the core level (either factually or by virtue of `force == true`), so we know that the expectation is for the controlled process to die.

Reply:
Okay, I will remove the comment.